tests/server/join/join_fail/join_fail_test.go

```diff
@@ -44,3 +46,36 @@ func TestFailedPDJoinInStep1(t *testing.T) {
 	re.Contains(err.Error(), "join failed")
 	re.NoError(failpoint.Disable("github.com/tikv/pd/server/join/addMemberFailed"))
 }
+
+func TestFailedToStartPDAfterSuccessfulJoin(t *testing.T) {
+	re := require.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	cluster, err := tests.NewTestCluster(ctx, 1)
+	defer cluster.Destroy()
+	re.NoError(err)
+
+	re.NoError(cluster.RunInitialServers())
+	re.NotEmpty(cluster.WaitLeader())
+
+	client := cluster.GetServer("pd1").GetEtcdClient()
+	pd2PeerURL := tempurl.Alloc()
+	// Add a member to the cluster but don't start it to simulate a successful join with failed start.
+	resp, err := client.MemberAdd(ctx, []string{pd2PeerURL})
+	re.NoError(err)
+	re.Len(resp.Members, 2)
+
+	// Join the second PD and start it.
+	pd2, err := cluster.Join(ctx, func(conf *config.Config, _ string) {
+		conf.AdvertisePeerUrls = pd2PeerURL
+	})
+	re.NoError(err)
+	re.NoError(pd2.Run())
+	re.NotEmpty(cluster.WaitLeader())
+
+	// Check that the new PD has joined the cluster.
+	members, err := client.MemberList(ctx)
+	re.NoError(err)
+	re.Len(members.Members, 2)
+	re.Equal(cluster.GetServer("pd1").GetClusterID(), pd2.GetClusterID())
```

Comment:
How about transferring the leadership to the newly added member to ensure it can serve as well?